lib/Parser/JSParserImpl.cpp

```diff
@@ -4579,15 +4624,20 @@ Optional<ESTree::ClassBodyNode *> JSParserImpl::parseClassBody(SMLoc startLoc) {
         break;
 
       case TokenKind::rw_static:
-        // static MethodDefinition
-        // static FieldDefinition
-        isStatic = true;
-        advance();
+        if (context_.getParseTS() && isStatic) {
+          // when `static` is already seen, treat the current one as identifier.
+          lexer_.unreserveCurrentToken();
```

Comment:
I think you've uncovered a more wide-reaching problem in the Hermes parser  relating to the use of `static` as a field name. Before class props were introduced, there was a simple and limited set of tokens after `static` that would indicate it was the method name instead of a modifier but now it is valid to have `static` be the name of the property as well.

This will require changes to work for Flow, JS, and TS. I will put up a fix and link you to it when it lands - that should allow you to simplify logic like this because everything regarding the reswords should be able to be properly handled in the call below.

I need to think about this more, but I'm not sure yet whether we'll end up needing `unreserveCurrentToken` or something like it - it's likely we won't because we do have logic in `parsePropertyName` to handle reswords and pull the identifiers out of them (no mutation of the `Token` itself needed).

Reply:
I've added https://github.com/facebook/hermes/commit/0d012688c8546bc927f3a011c02e6394c0f88fbf to make `static` parse properly, along with some tests.

Reply:
I've updated this PR based on the commit.
* Removed `unreserveCurrentToken` and `getAccessibilityIdent`.
* Moved `canNextTokenFollowModifier` to `JSParserImpl:: canFollowModifierTS`, along with the doc-comment.